src/core/moenda.js

```diff
@@ -0,0 +1,38 @@
+const path = require('path');
+const {translateOptions} = require(path.resolve(__dirname, '../loader'));
+const Linter = require(path.resolve(__dirname, './linter.js'));
+
+class Moenda {
+  constructor(options) {
+    this.results = [];
+    this.config = translateOptions(options);
+    this.linter = new Linter(
+      this.config.rules,
+      this.config.processor,
+      this.config.parser,
+    );
+  }
+
+  reporter(ruleName, filePath) {
+    return (payload) => {
+      this.results.push({
+        ruleName,
+        filePath,
+        column: 0,
+        ...payload,
+      });
+    };
+  }
+
+  runRules() {
```

Comment:
should I use promises here? I'm thinking if this could lead a problem if we have many files and rules

Reply:
no need to worry now.

first we need to create a basic structure for our project, then you create a test scenario with many files and rules, to allow us to test performance.